Fill.py

```diff
@@ -128,6 +129,34 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations:
     itempool.extend(unplaced_items)
 
 
+def accessibility_corrections(world: MultiWorld, state: CollectionState, locations, pool=[]):
+    maximum_exploration_state = sweep_from_pool(state, pool)
+    minimal_players = {player for player in world.player_ids if world.accessibility[player] == "minimal"}
+    unreachable_locations = [location for location in world.get_locations() if location.player in minimal_players and
+                             not location.can_reach(maximum_exploration_state)]
+    for location in unreachable_locations:
+        if (location.item is not None and location.item.classification == ItemClassification.progression and
+                location.address is not None and not location.locked and location.item.player not in minimal_players):
+            pool.append(location.item)
+            state.remove(location.item)
+            location.item = None
+            location.event = False
+            if location in state.events:
+                state.events.remove(location)
+            locations.append(location)
+
+    if pool:
+        fill_restrictive(world, state, locations, pool)
+
+
+def inaccessible_location_rules(world: MultiWorld, state: CollectionState, locations):
+    maximum_exploration_state = sweep_from_pool(state, [])
+    unreachable_locations = [location for location in locations if not location.can_reach(maximum_exploration_state)]
+    for location in unreachable_locations:
+        add_item_rule(location, lambda item: not ((item.useful or item.advancement) and
```

Comment:
`item.useful or item.advancement` can be expressed as the respective bit flags for some efficiency improvement.

Reply:
changed to `item.classification & 0b0011`